src/idl_gen_cpp.cpp

```diff
@@ -2198,14 +2215,26 @@ class CppGenerator : public BaseGenerator {
           } else if (field.value.type.base_type == BASE_TYPE_VECTOR) {
             code_ += "  auto {{FIELD_NAME}}__ = {{FIELD_NAME}} ? \\";
             const auto vtype = field.value.type.VectorType();
+            bool has_key = TypeHasKey(vtype);
             if (IsStruct(vtype)) {
               const auto type = WrapInNameSpace(*vtype.struct_def);
-              code_ += "_fbb.CreateVectorOfStructs<" + type + ">\\";
+              if (has_key) {
+                code_ += "_fbb.CreateVectorOfSortedStructs<" + type + ">\\";
+              } else {
+                code_ += "_fbb.CreateVectorOfStructs<" + type + ">\\";
+              }
```

Comment:
We can try ternary operator for readability, like this:
```cpp
code + = has_key ? "_fbb.CreateVectorOfSortedStructs" : "_fbb.CreateVectorOfStructs";
code += "<" + type + ">\\";
```

Reply:
please note that CodeWriter adds a newline every time you use operator+=

Reply:
Thank you for improvements.

The project has `.clang-format` format file, is CodeWriter use this file?
The  `clang-format` returns (format selected code only):
```c++
            if (IsStruct(vtype)) {
              const auto type = WrapInNameSpace(*vtype.struct_def);
              code_ += has_key ? "_fbb.CreateVectorOfSortedStructs"
                               : "_fbb.CreateVectorOfStructs";
              code_ += "<" + type + ">\\";
            } else if (has_key) {
              const auto type = WrapInNameSpace(*vtype.struct_def);
              code_ += "_fbb.CreateVectorOfSortedTables<" + type + ">\\";
            } else {
              const auto type = GenTypeWire(vtype, "", false);
              code_ += "_fbb.CreateVector<" + type + ">\\";
            }
            code_ +=
                has_key ? "({{FIELD_NAME}}) : 0;" : "(*{{FIELD_NAME}}) : 0;";
```
This is well-formated code without extra newline after `+=`.

Reply:
The line will exceed 80 characters - so the IDE decides to add a newline according .clang-format

Reply:
ok, thank you